components/projects/NewProject/NewProjectForm.tsx

```diff
@@ -29,9 +30,32 @@ function NewProjectForm({ employees, customers }: NewProjectFormProps): JSX.Elem
         }
     };
 
-    const handleSumbit = (e: React.FormEvent<HTMLFormElement>) => {
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
         e.preventDefault();
-        console.log(state);
+        // console.log(state);
+        const data = {
+            id: state.id,
+            name: state.name,
+            description: state.description,
+            startDate: state.startDate,
+            endData: state.endDate,
+            customer: { id: state.customer?.id, name: state.customer?.name },
+            managingEmployee: {
+                id: state.managingEmployee?.id,
+                first_name: state.managingEmployee?.first_name,
+                last_name: state.managingEmployee?.last_name,
+                email: state.managingEmployee?.email,
+            },
+        };
+        const url = `${process.env.NEXT_PUBLIC_API_URL}/project`;
+        const res = await fetch(url, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify(data),
```

Comment:
Can't you use just the `data` as is.

Reply:
If you mean omitting the `JSON.stringify` function then no. The backend will throw the following error:

`JSON parse error: Cannot deserialize value of type `three.consulting.epoc.dto.ProjectDTO` from Array value (token `JsonToken.START_ARRAY`); nested exception is com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `three.consulting.epoc.dto.ProjectDTO` from Array value (token `JsonToken.START_ARRAY`)<LF> at [Source: (PushbackInputStream); line: 1, column: 1]]`

Reply:
We also want some kind of response to both a successful and an unsuccessful post request. A response object has a key called `status` indicating its... well, status. For a successful `POST` request, this value will be `201` (see MDN documentation for [status codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) and for the fetch [response object](https://developer.mozilla.org/en-US/docs/Web/API/Response))

One can then write an if/else clause after `res` that checks the status code. If it's `201` we should change the state of `FormStatus` to be `SUCCESSFUL` and direct the user to the `/projects` page. You can do this by using the `router` object in Next, and its push function (see [here](https://nextjs.org/docs/api-reference/next/router))

If the server responds with anything else than a `201`, set the `FormStatus` field to `ERROR`. Additionally, add an error indicator to the JSX-code that only renders if the `FormStatus` is `ERROR`. It should tell the user that submitting the form failed.

Reply:
This is done in 5bf47e0. Succesful POST request sends user back to /projects (status 200 or 201). Any other status is handled by rendering a Box below Submit button displaying some text. What should happen in case of error should be elaborated.

Reply:
![Screenshot from 2021-11-09 15-50-27](https://user-images.githubusercontent.com/17109116/140936085-b46d28f5-803c-417d-8573-5587a5fa83a8.png)